docs/upgrading.md

```diff
@@ -0,0 +1,13 @@
+# Upgrading your installed application
+
+Upgrading the application is **not supported**.
+
+Fleet UI does not allow to upgrade an Elastic Agent running under Kubernetes, as that would conflict with the orchestration performed by Kubernetes itself.
+
+GCP Marketplace does not support upgrading an application via its UI.
+
+## Workaround
```

Comment:
nit: i think we could improve the wording a little here. for example, upgrading the application isn't supported _in fleet_. and i don't think the other methods here should be considered workarounds, since they are the recommended way to upgrade. 

do you think out of the two options presented below we should recommend one over the other?

Reply:
> upgrading the application isn't supported _in fleet_.

Is actually not supported in GCP Marketplace too.

> i don't think the other methods here should be considered workarounds, since they are the recommended way to upgrade.

great observation, I will update the wording here to remove the current negative connotation.

> do you think out of the two options presented below we should recommend one over the other?

(1) is the usual K8s process but I really think is just a matter of preference (and availability of the enrollment token, needed for (2)). I will reword this to clarify.

Reply:
updated in [`de3efcd` (#1)](https://github.com/elastic/elastic-agent-gcp-kubernetes-app/pull/1/commits/de3efcdea01e14f062af2ac990c9bb4a184820d6)

Reply:
looks great@